rootex/core/physics/debug_drawer.cpp

```diff
@@ -13,7 +13,7 @@ void DebugDrawer::drawLine(const btVector3& from, const btVector3& to, const btV
 		from.x(), from.y(), from.z(),
 	    to.x(), to.y(), to.z() });
 	IndexBuffer ib({ 0, 1 });
-	dynamic_cast<ColorMaterial*>(m_ColorMaterial.get())->setColor({ color.x(), color.y(), color.z(), 1.0f });
+	dynamic_cast<BasicMaterial*>(m_ColorMaterial.get())->setColor({ color.x(), color.y(), color.z(), 1.0f });
```

Comment:
The dynamic_cast here is not needed here

Reply:
it is of material type, hence needs to be cast before use. As to why this is done in such a manner? Because that's just how Material is being treated everywhere else. Also I have a feeling that there is no need to `setColor()` here, can you have a look @IronicallySerious ?

Reply:
Make the material directly of a Basic material type. Then no need for a cast every frame, just once in the constructor